components/experimental/src/units/ratio.rs

```diff
@@ -195,24 +208,184 @@ impl From<u32> for IcuRatio {
+            let denominator = components.next();
+            if components.next().is_some() {
+                return Err(IcuRatioError::MultipleSlashes);
+            }
+
+            let numerator = match numerator {
+                Some(numerator) => parse_decimal(numerator)?,
+                None => IcuRatio::zero(),
+            };
+
+            let denominator = match denominator {
+                Some(denominator) => parse_decimal(denominator)?,
+                None => IcuRatio::one(),
+            };
+
+            if denominator.is_zero() {
+                return Err(IcuRatioError::DivisionByZero);
+            }
+
+            Ok(numerator / denominator)
+        }
+
+        /// Parses a decimal number from a string input.
+        /// Accepts input in various decimal formats, including:
+        /// - Whole numbers: "1"
+        /// - Decimal numbers: "1.5"
+        /// An empty string input is interpreted as "0".
+        /// Note: Fractional inputs are not supported in this context.
+        fn parse_decimal(decimal: &str) -> Result<IcuRatio, IcuRatioError> {
+            let dot_count = decimal.matches('.').count();
+            match dot_count {
+                0 => BigRational::from_str(decimal)
+                    .map(IcuRatio)
+                    .map_err(|_| IcuRatioError::InvalidRatioString),
+                1 => {
+                    let decimal_part_length = decimal.split('.').last().unwrap().len();
+                    // remove "." from the string
+                    let number_str = decimal.replace(".", "");
+                    if number_str.is_empty() {
+                        return Err(IcuRatioError::InvalidRatioString);
+                    }
+                    let number = IcuRatio(
+                        BigRational::from_str(&number_str)
+                            .map_err(|_| IcuRatioError::InvalidRatioString)?,
+                    );
+                    let ten = IcuRatio::ten();
+                    let ten = ten.pow(decimal_part_length as i32);
+
+                    Ok(number / ten)
+                }
+                _ => Err(IcuRatioError::MultipleDecimalPoints),
+            }
+        }
+
+        let number_str = number_str.replace(",", ""); // Remove commas for parsing
+        if number_str.is_empty() {
+            return Ok(IcuRatio::zero());
+        }
+
+        let mut parts = number_str.split(|c| c == 'e' || c == 'E');
+        let significand = parts.next();
+        let exponent = parts.next();
         if parts.next().is_some() {
-            return Err(IcuRatioError::InvalidRatioString);
+            return Err(IcuRatioError::MultipleScientificNotations);
         }
 
-        let numerator = match numerator {
-            Some(num_str) => BigInt::from_str(num_str).map_err(IcuRatioError::BigIntParseError)?,
-            None => BigInt::zero(),
+        let significand = match significand {
+            Some(significand) => parse_fraction(significand)?,
```

Comment:
is the significant allowed to be a fraction when there's an exponent?

Reply:
Yes, we can have `1/2E+5`

Reply:
please add this to the docs

Reply:
done